### DISCUSSION TITLE
debug socket connection restarting on initial load

### DISCUSSION THREAD

**Discussion by simplecommerce** [vikejs/vike#1240] [Discussion] [answered]
Hi,

I have a weird issue and I don't know to debug it and hope you have any hints that could help.

When I start my test project, I also setup a websocket connection to my graphql server in order to subscribe to a subscription.
This works fine in dev but when I build for production, the initial load/connection makes my connection drop and then reconnect.

Here is the console log that shows this happening on my build:
 
```
npm run server:prod

> backend-core@3.0.0 server:prod
> cross-env NODE_ENV=production ts-node ./server/index.ts

Server running at http://localhost:3000
introspectionQuery done.
Creating Websocket Server
Creating wsClient. https://mysite.com/4000
wsClient attempting to connect to GraphQL subscription server.
wsClient connected to GraphQL subscription server.
wsClient connection error to GraphQL subscription server. <---- initial load here
Creating wsClient. https://mysite.com:4000
wsClient attempting to connect to GraphQL subscription server.
wsClient connected to GraphQL subscription server.
```

In the dev environment I never get his problem.
So I am unsure how to troubleshoot this behavior.

Any help is appreciated, thanks!

**UPDATE** Ok so, I wasn't able to figure out what the issue with my code was, but I decided to do a test, I created a child process with my GraphQL client subscription and did the same test again trying to load the site, and this time it doesn't close/reconnect my client subscription on initial load. So something in the process si causing my connection to close, probably due to the loading taking at least 10 seconds maybe.

Ok so I did a setInterval test, to see if I could print to my terminal the date/time to see if anything happens when the initial request comes, I set it to 1 sec.

```
pre renderPage     <------ when renderPage is called in server/index.ts
2023-11-05T15:49:56.359Z
2023-11-05T15:49:57.972Z
2023-11-05T15:50:03.334Z 
2023-11-05T15:50:20.623Z <-------- 20 secs later
2023-11-05T15:50:21.693Z
renderPage
pre renderPage
renderPage
2023-11-05T15:50:22.693Z
```

So we can see that something is blocking I assume the event loop causing my socket connection to close.
Having it as a separate child process works for now, but ideally I would have to figure out what is causing the initial load to take so much time and blocks the event loop.

**Comment by brillout** [comment, marked as answer]
I don't think this is related to Vike?

**Reply by simplecommerce** [reply]
Yeah you are right, I just did a small repo using the boiletplate ts + react and I am unable to reproduce it and it loads rapidly.
Its definitely something in my code, just need to figure out what makes it so slow to load initially.
